CMakeLists.txt

```diff
@@ -183,48 +183,48 @@ set(GOST_ENGINE_SOURCE_FILES
         )
 
 add_executable(test_digest test_digest.c)
-target_link_libraries(test_digest gost_engine gost_core ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_digest gost_engine gost_core ${OPENSSL_CRYPTO_LIBRARY} dl z pthread)
```

Comment:
Uhmmmmmmmmm, this shouldn't be necessary, anywhere.  the `${OPENSSL_CRYPTO_LIBRARY}` should already have those libraries linked to it.

However, looking at https://github.com/Kitware/CMake/blob/fa68fef6a719fcc150eac1eb8040d884f693d3dd/Modules/FindOpenSSL.cmake#L102-L111, it seems that this linkage is only done on Linux!
[sigh]

On what platform did you run into problems?

Reply:
I will try to explain.
I ran into the problems on Linux. I have OpenSSL build statically too. That's why I thought it might happen.
Those libraries exists in pkg_config file for openssl's libcrypto, but they are in `Libs.private` section. Like on the image below(there is no libz because I experimented with some flags for building OpenSSL).
![image](https://user-images.githubusercontent.com/10552394/129692802-5cd7b1cd-e9d0-4bc1-886e-5356fba1c66f.png)

Reply:
The point is that `find_package(OpenSSL 3.0 REQUIRED)` should have arranged it so you don't have to add those extra libraries...

Reply:
Now that I've finally understood that you're look at the openssl 1.1.1 branch, maybe I should stop looking at master :wink: